src/packages/shared-utils/user-helper.ts

```diff
@@ -42,28 +42,5 @@ const cognitoIdentitiesSchema = z.array(
     providerName: z.string(),
     providerType: z.string(),
     userId: z.string(),
-  }),
+  })
 );
-/** Takes the nullable string from CognitoUserAttributes.identities and parses is
- * to determine if a user is an IDM user or not. */
-export const isIDM = (identities: CognitoUserAttributes["identities"]) => {
-  if (!identities) return false;
-  try {
-    const parsedIdentities = cognitoIdentitiesSchema.parse(
-      JSON.parse(identities),
-    );
-    return parsedIdentities.some((identity) => identity.providerName === "IDM");
-  } catch (err: unknown) {
-    let message;
-    let issues;
-    if (err instanceof z.ZodError) {
-      issues = err.issues;
-      message = `Encountered Zod parse issues(${issues.length}): `;
-    } else if (err instanceof SyntaxError) {
-      issues = err.message;
-      message = `Encountered JSON parsing issue: `;
-    }
-    console.error(message, issues);
-    return false;
-  }
-};
```

Comment:
It won't let me apply it as a suggested edit, but with the removal of this and the tests associated, you're creating a niche case to account for in the IssueRai rule tests to verify we're properly identifying an IDM user. Keeping the tests would mean we have coverage wherever we need to check IDM users in the future.

We can keep the relevant tests and user helper function, but simplify it:

```typescript
const isIDM = (user: CognitoUserAttributes | null) => user?.username && !user.username.startsWith("IDM_")
```

The only tests that'd be removed after that would be the Zod and JSON assertions for error handling.